pkg/certificate/manager.go

```diff
@@ -112,9 +128,15 @@ func (m *Manager) getFromCache(cn CommonName) *Certificate {
 }
 
 // IssueCertificate implements Manager and returns a newly issued certificate from the given client.
-func (m *Manager) IssueCertificate(cn CommonName, validityPeriod time.Duration) (*Certificate, error) {
+func (m *Manager) IssueCertificate(prefix string, opts ...IssueOption) (*Certificate, error) {
 	var err error
-	cert := m.getFromCache(cn) // Don't call this while holding the lock
+	cert := m.getFromCache(prefix) // Don't call this while holding the lock
+
+	options := defaultOptions(m.serviceCertValidityDuration)
```

Comment:
So there's 2 types here, an `IssueOption`, which is a type alias for a function that operates on a `type issueOptions struct`

The manager has a service validity duration to leverage specifically for rotation, as it doesn't have that info at rotation time otherwise. Here, we create an `issueOptions` type, using the default validity duration, and then apply all of the `IssueOption`, which could override the validity period to something else.

Note the only exported type is IssueOption, so the defaultOptions/issueOptions are a hidden implementation detail.